exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/ColumnType.kt

```diff
@@ -794,13 +793,16 @@ class BlobColumnType : ColumnType() {
     }
 
     override fun readObject(rs: ResultSet, index: Int) = when {
-        currentDialect is SQLServerDialect -> rs.getBytes(index)?.let(::ExposedBlob)
-        else -> rs.getBinaryStream(index)?.let(::ExposedBlob)
+        currentDialect is SQLServerDialect || currentDialect is SQLiteDialect ->
+            rs.getBytes(index)?.let(::ExposedBlob)
+
+        else -> rs.getBlob(index)?.binaryStream?.let(::ExposedBlob)
```

Comment:
Please add a property to the class and isolate only the necessary changes to PostgreSQL only:
```kt
class BlobColumnType(
    /** Returns whether an OID column should be used instead of BYTEA. This value only applies to PostgreSQL databases. */
    val useObjectIdentifier: Boolean = false
) : ColumnType() {
    override fun sqlType(): String = when {
        useObjectIdentifier && currentDialect is PostgreSQLDialect -> "oid"
        useObjectIdentifier -> error("Storing BLOBs using OID columns is only supported by PostgreSQL")
        else -> currentDialect.dataTypeProvider.blobType()
    }

    override fun readObject(rs: ResultSet, index: Int) = when {
        currentDialect is SQLServerDialect -> rs.getBytes(index)?.let(::ExposedBlob)
        currentDialect is PostgreSQLDialect && useObjectIdentifier -> rs.getBlob(index)?.binaryStream?.let(::ExposedBlob)
        else -> rs.getBinaryStream(index)?.let(::ExposedBlob)
    }
}
```

The following files will then require changes to enable this new property:
```kt
// Table.kt - edit blob() and its KDocs

/**
  * Creates a binary column, with the specified [name], for storing BLOBs.
  *
  * Setting [useObjectIdentifier] to `true` creates an OID column instead of a BYTEA column and is only
  * supported by PostgreSQL dialects.
  *
  * @sample org.jetbrains.exposed.sql.tests.shared.DDLTests.testBlobAsOid
  */
fun blob(name: String, useObjectIdentifier: Boolean = false): Column<ExposedBlob> =
    registerColumn(name, BlobColumnType(useObjectIdentifier))

// Op.kt - edit blobParam() and its KDocs

/**
 * Returns the specified [value] as a blob query parameter.
 *
 * Set [useObjectIdentifier] to `true` if the parameter should be processed using an OID column instead of a
 * BYTEA column. This is only supported by PostgreSQL databases.
 */
fun blobParam(value: ExposedBlob, useObjectIdentifier: Boolean = false): Expression<ExposedBlob> =
    QueryParameter(value, BlobColumnType(useObjectIdentifier))
```